Sources/TuistCloud/Client/AuthenticationMiddleware.swift

```diff
@@ -0,0 +1,33 @@
+import OpenAPIRuntime
+import Foundation
+import TuistSupport
+
+/// Injects an authorization header to every request.
+struct AuthenticationMiddleware: ClientMiddleware {
+    func intercept(
+        _ request: Request,
+        baseURL: URL,
+        operationID: String,
+        next: (Request, URL) async throws -> Response
+    ) async throws -> Response {
+        var request = request
+        let environment = ProcessInfo.processInfo.environment
+        let tokenFromEnvironment = environment[Constants.EnvironmentVariables.cloudToken]
+        let token: String?
+        if CIChecker().isCI() {
+            token = tokenFromEnvironment
+        } else {
+            token = try? tokenFromEnvironment ?? CredentialsStore().read(serverURL: baseURL)?.token
+        }
+
+        if let token = token {
+            request.headerFields.append(.init(
+                name: "Authorization", value: "Bearer \(token)"
+            ))
+        } else {
+            try CloudSessionController().authenticate(serverURL: baseURL)
+            // TODO: Fix
+        }
```

Comment:
I'd not try to authenticate if the token is not present and instead fail telling users how they can authenticate. When we start adding these possible paths to existing workflows users will end up wanting to control it, which leads to commands that are harder to reason about.

Reply:
Sure, I like that suggestion!